### ISSUE TITLE
Bittorrent protocol stream over websocket

### ISSUE THREAD

**Issue by binux** [feross/webtorrent-hybrid#16]
Thanks for the great projects! I've made some P2P video sharing via WebRTC in 2013 with some crude protocol, a real bittorrent protocol over webrtc which is great!

Now I'm trying to make something that can download resources in browser from normal BT network by using webtorrent-hybrid. But find wrtc is not stable, `node` is crashed several times during a download.

I thinking about using bittorrent protocol over [websocket stream](https://github.com/maxogden/websocket-stream) like: `websocket_stream.pipe(wire).pipe(websocket_stream)`. As webtorrent-hybrid is running on kind of node server.

--------------
By checking the source code, I noticed some modules needs to be modified:

* adding a websocket Peer, a websocket pool to bittorrent-swarm
* bittorrent-tracker need websocket peer broadcast support
* submit websocket address information via torrent-discovery

As there are a lot hard coded peer type detection and `_onWebSocketRequest` stuff, so I'm asking if you can consider adding this websocket peer.

Happy New Year!

**Comment by feross** [comment]
Hi @binux, thanks for opening such a thoughtful issue.

I think this is out-of-scope for WebTorrent, unfortunately. I'm hoping that we'll soon have a reliable webrtc implementation on the server-side ([this one](https://github.com/nickdesaulniers/node-rtc-peer-connection) looks promising) so we won't need hacks like using a websocket proxy server.

A solution using a websocket proxy server isn't true P2P, since you're running a server that sits between the browsers and the normal bittorrent peers. It's a temporary hack at best, and that's why I don't think it makes sense to put in the effort to support it. You're free to fork the code and implement it for your own uses though :)

Cheers!

**Comment by binux** [comment]
I'm not suggesting a websocket proxy, it's still true bittorrent protocol, just over websocket not tcp or RTCPeerConnection. bittorrent-hybrid or any other clients that implemented bittorrent protocol over websocket can connect to it and working as true P2P.

I have already implemented the `WebSocketPeer` stuff, but soon noticed I cannot announce the address. I planning announce peer id like `ws://ip:port/` but find a lot of hard coded peer type detection. As a lot of hack is needed through multiple modules, I opened this issue.

Anyway, it is true that BT over websocket only work between bittorrent-hybrid peers and browser peers, if there is a reliable webrtc implementation, it is not needed. Maybe will come back some month later.

Thank you for your work again :+1:
